crates/gpui/src/platform/windows/window.rs

```diff
@@ -0,0 +1,536 @@
+    active_status_change: Option<Box<dyn FnMut(bool)>>,
+    resize: Option<Box<dyn FnMut(Size<Pixels>, f32)>>,
+    fullscreen: Option<Box<dyn FnMut(bool)>>,
+    moved: Option<Box<dyn FnMut()>>,
+    should_close: Option<Box<dyn FnMut() -> bool>>,
+    close: Option<Box<dyn FnOnce()>>,
+    appearance_changed: Option<Box<dyn FnMut()>>,
+}
+
+pub(crate) struct WindowsWindow {
+    inner: Rc<WindowsWindowInner>,
+}
+
+struct WindowCreateContext {
+    inner: Option<Rc<WindowsWindowInner>>,
+    platform_inner: Rc<WindowsPlatformInner>,
+    handle: AnyWindowHandle,
+}
+
+impl WindowsWindow {
+    pub(crate) fn new(
+        platform_inner: Rc<WindowsPlatformInner>,
+        handle: AnyWindowHandle,
+        options: WindowOptions,
+    ) -> Self {
+        let dwexstyle = WINDOW_EX_STYLE::default();
+        let classname = register_wnd_class();
+        let windowname = HSTRING::from(
+            options
+                .titlebar
+                .as_ref()
+                .and_then(|titlebar| titlebar.title.as_ref())
+                .map(|title| title.as_ref())
+                .unwrap_or(""),
+        );
+        let dwstyle = WS_OVERLAPPEDWINDOW & !WS_VISIBLE;
+        let mut x = CW_USEDEFAULT;
+        let mut y = CW_USEDEFAULT;
+        let mut nwidth = CW_USEDEFAULT;
+        let mut nheight = CW_USEDEFAULT;
+        match options.bounds {
+            WindowBounds::Fullscreen => {}
+            WindowBounds::Maximized => {}
+            WindowBounds::Fixed(bounds) => {
+                x = bounds.origin.x.0 as i32;
+                y = bounds.origin.y.0 as i32;
+                nwidth = bounds.size.width.0 as i32;
+                nheight = bounds.size.height.0 as i32;
+            }
+        };
+        let hwndparent = HWND::default();
+        let hmenu = HMENU::default();
+        let hinstance = HINSTANCE::default();
+        let mut context = WindowCreateContext {
+            inner: None,
+            platform_inner: platform_inner.clone(),
+            handle,
+        };
+        let lpparam = Some(&context as *const _ as *const _);
+        unsafe {
+            CreateWindowExW(
+                dwexstyle,
+                classname,
+                &windowname,
+                dwstyle,
+                x,
+                y,
+                nwidth,
+                nheight,
+                hwndparent,
+                hmenu,
+                hinstance,
+                lpparam,
+            )
+        };
+        let wnd = Self {
+            inner: context.inner.unwrap(),
+        };
+        platform_inner.window_handles.borrow_mut().insert(handle);
+        match options.bounds {
+            WindowBounds::Fullscreen => wnd.toggle_full_screen(),
+            WindowBounds::Maximized => wnd.maximize(),
+            WindowBounds::Fixed(_) => {}
+        }
+        unsafe { ShowWindow(wnd.inner.hwnd, SW_SHOW) };
+        wnd
+    }
+
+    fn maximize(&self) {
+        log::debug!("{:?}", self.inner.hwnd);
```

Comment:
leftover from debugging? If we are to keep this, we should at least tell that it's a maximization event

Reply:
I removed it